src/data/markdown/docs/07 extensions/03 Guides/build-k6-using-docker.md

```diff
@@ -0,0 +1,123 @@
+---
+title: 'Build a k6 binary using Docker'
+excerpt: ''
+hideFromSidebar: false
+---
+
+The easiest way to use [xk6](https://github.com/grafana/xk6) is via our [Docker image](https://hub.docker.com/r/grafana/xk6/). This avoids having to setup a local Go environment, and install xk6 manually.
+
+## Building your first extension
+
+For example, to build a k6 v0.43.1 binary with the [`xk6-kafka`](https://github.com/mostafa/xk6-kafka) and [`xk6-output-influxdb`](https://github.com/grafana/xk6-output-influxdb) extensions, you would run one of the following based upon your operating system:
+
+<CodeGroup labels={["Linux", "Mac", "Windows PowerShell", "Windows"]}>
+
+```bash
+docker run --rm -u "$(id -u):$(id -g)" -v "${PWD}:/xk6" grafana/xk6 build v0.43.1 \
+  --with github.com/mostafa/xk6-kafka@v0.17.0 \
+  --with github.com/grafana/xk6-output-influxdb@v0.3.0
+```
+
+```bash
+docker run --rm -e GOOS=darwin -u "$(id -u):$(id -g)" -v "${PWD}:/xk6" \
+  grafana/xk6 build v0.43.1 \
+  --with github.com/mostafa/xk6-kafka@v0.17.0 \
+  --with github.com/grafana/xk6-output-influxdb@v0.3.0
+```
+
+```powershell
+docker run --rm -e GOOS=windows -u "$(id -u):$(id -g)" -v "${PWD}:/xk6" `
+  grafana/xk6 build v0.43.1 --output k6.exe `
+  --with github.com/mostafa/xk6-kafka@v0.17.0 `
+  --with github.com/grafana/xk6-output-influxdb@v0.3.0
+```
+
+```batch
+docker run --rm -e GOOS=windows -v "%cd%:/xk6" ^
+  grafana/xk6 build v0.43.1 --output k6.exe ^
+  --with github.com/mostafa/xk6-kafka@v0.17.0 ^
+  --with github.com/grafana/xk6-output-influxdb@v0.3.0
+```
+
+</CodeGroup>
+
+This would create a `k6` (or `k6.exe`) binary in the current working directory.
+
+## Breaking down the command
+
+We'll admit, the example command line looks terrifying. Let's focus on the first part, which pertains strictly to Docker:
+
+```bash
+docker run --rm -u "$(id -u):$(id -g)" -v "${PWD}:/xk6"
+```
+
+This tells Docker to run a new container from an image. 
+`--rm` means the container will be destroyed once your build is completed.
+`-u` specifies the user and group IDs of the account on the host machine. This is important for the `k6` file to have the same file permissions as the host user.
+`-v` is required to mount the current working directory inside the container, so that the `k6` binary can be written to it.
```

Comment:
A list, for clarity?

```suggestion
- `--rm` means the container will be destroyed once your build is completed.
- `-u` specifies the user and group IDs of the account on the host machine. This is important for the `k6` file to have the same file permissions as the host user.
- `-v` is required to mount the current working directory inside the container, so that the `k6` binary can be written to it.
```

Side note: we should probably link to docker docs, not sure if to each section like https://docs.docker.com/storage/volumes/ for `-v`, as the link could break in the future?